pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -126,10 +126,15 @@ const (
 	maxSecretPrefixNameLength = maxNameLength - randomLength
 )
 
-func getDockercfgSecretNamePrefix(serviceAccount *api.ServiceAccount) string {
+func GetDockercfgSecretNamePrefix(serviceAccount *api.ServiceAccount) string {
 	return namer.GetName(serviceAccount.Name, "dockercfg-", maxSecretPrefixNameLength)
 }
-func getTokenSecretNamePrefix(serviceAccount *api.ServiceAccount) string {
+
+// GetTokenSecretNamePrefix creates the prefix used for the generated SA token secret. This is compatible with kube up until
```

Comment:
@liggitt @kargakis requested a refactor which makes sense, but if I'm already into refactoring, I'd prefer to fix the upstream code to create SA tokens like this.  Do you have an objection?